bin/exec.go

```diff
@@ -119,14 +119,37 @@ func main() {
 			log.Infoln("Unable to find ConfigMaps")
 		}
 
+
+		secretsExist, secrets := utils.CheckSecrets(engineDetails, config, engineDetails.Experiments[i])
+
+		var validatedSecrets []v1alpha1.Secret
+		var errorsListForSecrets []error
+		if secretsExist == true {
+			log.Infoln("Secrets Found")
+			//fetch details and apply those config maps needed
+			// to be used in the job creation
+			// first convert the format of ConfigMap's Data to map[string]string
+			// & then use the kube-builder to build config maps
+			validatedSecrets, errorsListForSecrets = utils.ValidateSecrets(secrets, engineDetails, clients)
+
+			//log.Infoln("Printing VolumeMounts : ", volumeMounts)
+			if errorsListForSecrets != nil {
+				log.Errorf("Printing Errors, found while Validating Secrets : %v, Will abort the Experiment Execution", errorsListForSecrets)
+				continue
+			}
+		} else {
+			log.Infoln("Unable to find Secrets")
+		}
+
 		log.Infof("Printing Validated ConfigMaps: %v", validatedConfigMaps)
+		log.Infof("Printing Validated Secrets: %v", validatedSecrets)
 
 		// 1. []*volume.Builder
-		volumeBuilders := utils.CreateVolumeBuilder(validatedConfigMaps)
+		volumeBuilders := utils.CreateVolumeBuilder(validatedConfigMaps, validatedSecrets)
```

Comment:
Sorry to pick again: I understand that CreateVolumeBuilder is a common fn needed for both configmaps and secrets. So why is it inside configMap utils?

Reply:
Yes, so where should I place it? Will place in some commonUtils!